configure.ac

```diff
@@ -80,6 +80,20 @@ AC_SUBST([AM_CPPFLAGS], ["$MY_CPPFLAGS \
                           -isystem\$(top_srcdir)/third_party/spdlog/include \
                           -I\$(top_srcdir)/modules/BMI/include"])
 
+# Thrift
```

Comment:
I am a little confused. I think you are asking the user to give you the path to the `thrift` compiler executable that he wants to use (either the apache one or the p4 one). But what about the thrift library? What if the wrong thrift library is picked, i.e. not the one corresponding to the `thrift` executable you want? Please correct me if I am missing something.
I believe that you should not make this a configure option. You should pick whichever comes first in the path (which is what is happening for the library right now). Instead of your `configure.ac` script should detect which library is being used and define the  `P4THRIFT` file if needed. I think th easier way to do this is to have a header file in your p4 thrift package that does not exist in the apache one. You can then do the detection with code similar to this:
```
AC_CHECK_HEADER([thrift/dummy_p4_thrift.h], [use_p4_thrift=yes], [use_p4_thrift=no])])
// one of these 2 lines I guess, depending on what you need, but I am guessing the second one
AM_CONDITIONAL([P4THRIFT], [test "$use_p4_thrift" = "yes"])
AS_IF([test "$use_p4_thrift" = "yes"], [MY_CPPFLAGS="$MY_CPPFLAGS -DP4THRIFT])
```